flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/JsonRowDataSerDeSchemaTest.java

```diff
@@ -428,84 +436,111 @@ private void testParseErrors(TestSpec spec) throws Exception {
 
 		TestSpec
 			.json("{\"id\":\"112.013.123\"}")
 			.rowType(ROW(FIELD("id", FLOAT())))
-			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"112.013.123\"}'"),
+			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"112.013.123\"}'."),
 
 		TestSpec
 			.json("{\"id\":\"112.013.123\"}")
 			.rowType(ROW(FIELD("id", DOUBLE())))
-			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"112.013.123\"}'"),
+			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"112.013.123\"}'."),
 
 		TestSpec
 			.json("{\"id\":\"18:00:243\"}")
 			.rowType(ROW(FIELD("id", TIME())))
-			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"18:00:243\"}'"),
+			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"18:00:243\"}'."),
 
 		TestSpec
 			.json("{\"id\":\"18:00:243\"}")
 			.rowType(ROW(FIELD("id", TIME())))
-			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"18:00:243\"}'"),
+			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"18:00:243\"}'."),
 
 		TestSpec
 			.json("{\"id\":\"20191112\"}")
 			.rowType(ROW(FIELD("id", DATE())))
-			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"20191112\"}'"),
+			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"20191112\"}'."),
 
 		TestSpec
 			.json("{\"id\":\"20191112\"}")
 			.rowType(ROW(FIELD("id", DATE())))
-			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"20191112\"}'"),
+			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"20191112\"}'."),
+
+		TestSpec
+			.json("{\"id\":true}")
+			.rowType(ROW(FIELD("id", STRING())))
+			.expect(Row.of("true")),
+
+		TestSpec
+			.json("{\"id\":123.234}")
+			.rowType(ROW(FIELD("id", STRING())))
+			.expect(Row.of("123.234")),
+
+		TestSpec
+			.json("{\"id\":1234567}")
+			.rowType(ROW(FIELD("id", STRING())))
+			.expect(Row.of("1234567")),
+
+		TestSpec
+			.json("{\"id\":\"string field\"}")
+			.rowType(ROW(FIELD("id", STRING())))
+			.expect(Row.of("string field")),
+
+		TestSpec
+			.json("{\"id\":[\"array data1\",\"array data2\",123,234.345]}")
+			.rowType(ROW(FIELD("id", STRING())))
+			.expect(Row.of("[\"array data1\",\"array data2\",123,234.345]")),
+
+		TestSpec
+			.json("{\"id\":{\"k1\":123,\"k2\":234.234,\"k3\":\"string data\"}}")
+			.rowType(ROW(FIELD("id", STRING())))
+			.expect(Row.of("{\"k1\":123,\"k2\":234.234,\"k3\":\"string data\"}")),
 
 		TestSpec
 			.json("{\"id\":\"2019-11-12 18:00:12\"}")
 			.rowType(ROW(FIELD("id", TIMESTAMP(0))))
 			.timestampFormat(TimestampFormat.ISO_8601)
-			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"2019-11-12 18:00:12\"}'"),
+			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"2019-11-12 18:00:12\"}'."),
 
 		TestSpec
 			.json("{\"id\":\"2019-11-12T18:00:12\"}")
 			.rowType(ROW(FIELD("id", TIMESTAMP(0))))
-			.expectErrorMessage("Failed to deserialize JSON '{\"id\":\"2019-11-12T18:00:12\"}'"),
+			.expect(Row.of(LocalDateTime.parse("2019-11-12T18:00:12"))),
```

Comment:
This should fail. We should update `testIgnoreParseErrors` to use `spec.timestampFormat` instead of hard code ISO.